### PR TITLE
fix: Allow mtls sts endpoint for external account token urls.

### PR DESCRIPTION
**Pull request by clundin25**

### CODE REVIEW

**Review by arithmetic1728 — APPROVED** [review]

File: tests/test_external_account.py [hunk 1]
```diff
@@ -67,6 +67,7 @@
 
 VALID_TOKEN_URLS = [
     "https://sts.googleapis.com",
+    "https://sts.mtls.googleapis.com",
```
Comment (lsirac) [inline tests/test_external_account.py:70]: Probably worth it to add some test cases

File: google/auth/external_account.py [hunk 2]
```diff
@@ -439,6 +439,7 @@ def validate_token_url(token_url, url_type="token"):
         _TOKEN_URL_PATTERNS = [
             "^[^\\.\\s\\/\\\\]+\\.sts\\.googleapis\\.com$",
             "^sts\\.googleapis\\.com$",
+            "^sts\\.mtls\\.googleapis\\.com$",
```
Comment (lsirac) [inline google/auth/external_account.py:442]: Do we need to support regional mtls endpoints?

Comment (clundin25) [reply google/auth/external_account.py:442]: Let me check if that is the case

File: tests/test_external_account.py (hunk 1 above)
Comment (clundin25) [reply tests/test_external_account.py:70]: This should get included in this loop here https://github.com/googleapis/google-auth-library-python/blob/690b05509b7679b1d32fe71f36760d0f8c739268/tests/test_external_account.py#L342

File: google/auth/external_account.py (hunk 2 above)
Comment (clundin25) [reply google/auth/external_account.py:442]: I can confirm regional MTLS endpoints should also be supported. Will follow up.

File: tests/test_external_account.py (hunk 1 above)
Comment (lsirac) [reply tests/test_external_account.py:70]: Let's add some invalid cases too

**Review by lsirac — APPROVED** [review]

File: google/auth/external_account.py [hunk 3]
```diff
@@ -437,11 +437,11 @@ def _initialize_impersonated_credentials(self):
     @staticmethod
     def validate_token_url(token_url, url_type="token"):
         _TOKEN_URL_PATTERNS = [
```
[the remark this answers is not in the document]
Comment (BigTailWolf) [reply google/auth/external_account.py:439]: Not all languages apply to raw strings. Here we are keeping all the languages using the same pattern.

**Review by BigTailWolf — APPROVED** [review]
LGTM

File: tests/test_external_account.py [hunk 4]
```diff
@@ -67,23 +67,31 @@
...
     "https://us-east-1.sts.googleapis.com",
+    "https://us-east-1.sts.mtls.googleapis.com",
     "https://US-EAST-1.sts.googleapis.com",
     "https://sts.us-east-1.googleapis.com",
     "https://sts.US-WEST-1.googleapis.com",
```
Comment (sai-sunder-s) [inline tests/test_external_account.py:75]: add a case for "https://sts.mtls.US-WEST-1.googleapis.com" (if that is allowed)?

Comment (clundin25) [reply tests/test_external_account.py:75]: IIUC MTLS must prefix googleapis.com but I will check

**Pull request merged**